File: rob599_basic/src/basic_node.py
#!/usr/bin/env python3

# Basic structure of a Python node.
#
# Bill Smart
#
# This example shows the basic structure of a ROS node in Python.  This node doesn't
# actually do anything, # but does contain all of the parts needed to set up a node.


# This pulls in the basic Python functionality for ROS.  You will need this in every
# ROS Python node.
import rospy

# It's often a good idea to also import sys, so that we can get access to any
# command-line parameters given to the node.
import sys


# Execution of the node starts here, since it's called as an executable.
if __name__ == '__main__':
	# All packages need to be initialized.  The only required argument here is a
	# name for the node.  We can also pass in any command-line parameters that are
	# fed to the node.
	rospy.init_node('basic_node', argv=sys.argv)

	# Give control over to ROS.  The node will sit here until it gets killed,
	# usually with a ctrl-c from the keyboard
	rospy.spin()

File: rob599_basic/src/basic_node.cpp
// The basic C++ node.
//
// basic_node.cpp
//
// Bill Smart
//
// This is an example of the most basic C++ node in ROS.  It doesn't actually do anything useful,
// but does illustrate some of the boilerplate code that you'll need in all of your other nodes.


// Include the basic ROS stuff.  This is the C++ equivalent to import rospy in Python nodes.
#include <ros/ros.h>


// This is the main function, where program execution starts.  This is similar to the
//  if __name__ == '__main__': line in Python.
int main(int argc, char **argv) {
	// Initualize the node.  Note that the argument order is different in C++ and Python.
	ros::init(argc, argv, "basic_node_cpp");

	// Every node needs to make a NodeHandle.
	ros::NodeHandle node;

	// Give control over to ROS.
	ros::spin();

	// Returning 0 to the operating system, signals that the code ended normally.
	return 0;
}